From 7e8e366cb5456df01c9abcd43cef291f0c75b402 Mon Sep 17 00:00:00 2001
From: setiawanjimmy <setiawanjimmy@gmail.com>
Date: Thu, 24 Nov 2016 08:58:50 +0100
Subject: [PATCH 2/6] Add missing permission

- Fixes PackageInstaller FC when tapping on a downloaded apk notification

11-24 05:30:39.780 E/AndroidRuntime(11820): FATAL EXCEPTION: AsyncTask #1
11-24 05:30:39.780 E/AndroidRuntime(11820): Process: com.android.packageinstaller, PID: 11820
11-24 05:30:39.780 E/AndroidRuntime(11820): java.lang.RuntimeException: An error occurred while executing doInBackground()
11-24 05:30:39.780 E/AndroidRuntime(11820): at android.os.AsyncTask.done(AsyncTask.java:318)
11-24 05:30:39.780 E/AndroidRuntime(11820): at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
11-24 05:30:39.780 E/AndroidRuntime(11820): at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
11-24 05:30:39.780 E/AndroidRuntime(11820): at java.util.concurrent.FutureTask.run(FutureTask.java:242)
11-24 05:30:39.780 E/AndroidRuntime(11820): at android.os.AsyncTask.run(AsyncTask.java:243)
11-24 05:30:39.780 E/AndroidRuntime(11820): at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
11-24 05:30:39.780 E/AndroidRuntime(11820): at java.util.concurrent.ThreadPoolExecutor.run(ThreadPoolExecutor.java:607)
11-24 05:30:39.780 E/AndroidRuntime(11820): at java.lang.Thread.run(Thread.java:761)
11-24 05:30:39.780 E/AndroidRuntime(11820): Caused by: java.lang.SecurityException: Permission Denial: reading com.android.providers.downloads.DownloadProvider uri content://downloads/all_downloads/101 from pid=11820, uid=10025 requires android.permission.ACCESS_ALL_DOWNLOADS, or grantUriPermission()

Change-Id: Ifa2d1d75ebf42e3b0d44f87ca1ed04d41ffeb394
---
 AndroidManifest.xml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 67e0b4cd..01444b20 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -25,6 +25,7 @@
     <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
     <uses-permission android:name="android.permission.HIDE_NON_SYSTEM_OVERLAY_WINDOWS" />
     <uses-permission android:name="android.permission.USE_RESERVED_DISK" />
+    <uses-permission android:name="android.permission.ACCESS_ALL_DOWNLOADS" />
 
     <uses-permission android:name="com.google.android.permission.INSTALL_WEARABLE_PACKAGES" />
 
-- 
2.17.1

